src/NodeAutoCompleteViewExtension/ViewModels/NodeAutoCompleteBarViewModel.cs

```diff
@@ -195,7 +190,7 @@ public int SelectedIndex
             }
             set
             {
-                if(selectedIndex != value && value >= 0)
+                if(selectedIndex != value && value >= 0 && selectedIndex != -1)
```

Comment:
Not sure about this condition selectedIndex != -1
I was expecting here to check the value to be between 0 and less than total number of clusters available.

Reply:
I would say to make this more clear. If for some reason someone sets the the selectedIndex to -1 the next real update will be missed because of this condition.
This is actually happening - we are setting the value to 0 - we don't add the node but then we manually add the cluster.

Reply:
For  switching to single mode, -1 is selected at first and it results in errors when readd node is called.

Reply:
Basically ReAdd node is being called twice when you switch modes. So if there is a better way to prevent that, I can do that instead